src/graphql/fetchAllRequestIdentifiers.ts

```diff
@@ -1,65 +1,118 @@
+import { IdentifierType } from '@transcend-io/privacy-types';
+import { decodeCodec } from '@transcend-io/type-utils';
+import type { Got } from 'got';
 import { GraphQLClient } from 'graphql-request';
+import * as t from 'io-ts';
 import { REQUEST_IDENTIFIERS } from './gqls';
 import { makeGraphQLRequest } from './makeGraphQLRequest';
-import { IdentifierType } from '@transcend-io/privacy-types';
 
-export interface RequestIdentifier {
+const literalUnion = <T extends t.Mixed>(
+  values: T[],
+): t.UnionC<[T, T, ...T[]]> => t.union(values as [T, T, ...T[]]);
+
+const IdentifierTypeValues = literalUnion(
+  Object.values(IdentifierType).map((v) => t.literal(v)),
+);
+
+const RequestIdentifier = t.type({
   /** ID of request */
-  id: string;
+  id: t.string,
   /** Name of identifier */
-  name: string;
+  name: t.string,
   /** The underlying identifier value */
-  value: string;
-  /** Identifier metadata */
-  identifier: {
-    /** Type of identifier */
-    type: IdentifierType;
-  };
+  value: t.string,
+  /** Type of identifier */
+  type: IdentifierTypeValues,
   /** Whether request identifier has been verified at least one */
-  isVerifiedAtLeastOnce: boolean;
+  isVerifiedAtLeastOnce: t.boolean,
   /** Whether request identifier has been verified */
-  isVerified: boolean;
-}
+  isVerified: t.boolean,
+});
+
+/** Type override */
+export type RequestIdentifier = t.TypeOf<typeof RequestIdentifier>;
 
 const PAGE_SIZE = 50;
 
+export const RequestIdentifiersResponse = t.type({
+  identifiers: t.array(RequestIdentifier),
+});
+
 /**
  * Fetch all request identifiers for a particular request
  *
- * @param client - GraphQL client
- * @param options - Filter options
+ * @param options - Options
  * @returns List of request identifiers
  */
-export async function fetchAllRequestIdentifiers(
-  client: GraphQLClient,
-  {
-    requestId,
-  }: {
-    /** ID of request to filter on */
-    requestId: string;
-  },
-): Promise<RequestIdentifier[]> {
+export async function fetchAllRequestIdentifiers({
+  client,
```

Comment:
small nit: most of the other functions in the package have the client as the first arg and all options as a second - may be good to maintain that same interface here

Reply:
So, client first, sombra second, then the options? Or put sombra in the options?

Reply:
ah good q - i'd prob put sombra second  arg, rest in options